### ISSUE TITLE
Support for private channels on Slack Triggers

### ISSUE THREAD

**Issue by kmai** [argoproj/argo-events#877]
**Is your feature request related to a problem? Please describe.**
Slack trigger should allow sending messages to private channels

**Describe the solution you'd like**
Taking the logic from https://github.com/argoproj/argo-events/blob/master/sensors/triggers/slack/slack.go#L109-L129:

```go
        params := &slack.GetConversationsParameters{
		Limit: 200,
	}
```

should become

```go
	params := &slack.GetConversationsParameters{
		Limit: 200,
		Types: []string{"public_channel", "private_channel"},
	}
```

to also support private channels.

**Describe alternatives you've considered**
Usually, the alternative would be to have a public channel, but people might want to avoid that for security reasons.

**Additional context**
* I believe this feature should be easy to implement (and we could also even add a parameter on the trigger definition to make this more customizable!). 
* Also, we should mention that the permissions to fetch channels using slack's `conversation.list` method requires `channels:read` or `groups:read` depending on the type of channel.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by VaibhavPage** [comment]
@kmai Would you like to work on this enhancement?

**Comment by kmai** [comment]
I was actually was trying to get a working build environment for Argo but the documentation on how to contribute is a bit outdated.

But yeah, I'd like to. I know I would have to update the type for the SlackTrigger so it can be defined on the CRD, but I'm stuck on running `make codegen` for some reason.

**Comment by shashwat-appdirect** [comment]
Hi @kmai,
This is already fixed and available in v1.2.2. Please check
